core/src/main/java/org/elasticsearch/action/admin/indices/recovery/RecoveryResponse.java

```diff
@@ -127,4 +129,13 @@ public void readFrom(StreamInput in) throws IOException {
             shardRecoveryStates.put(s, list);
         }
     }
+
+    @Override
+    public String toString() {
+        try {
+            return XContentHelper.toXContent(this, XContentType.JSON, true).utf8ToString();
```

Comment:
Strings.toString already catches IOException. there is a variant of it where you can pass in pretty and himanReadable flags.